engine/classes/Elgg/Cache/PluginSettingsCache.php

```diff
@@ -0,0 +1,166 @@
+<?php
+namespace Elgg\Cache;
+
+use Elgg\Database;
+
+/**
+ * In memory cache of (non-user-specific, non-internal) plugin settings
+ *
+ * @access private
+ */
+class PluginSettingsCache {
+
+	/**
+	 * The cached values.
+	 *
+	 * @var array GUID => string[]
+	 */
+	protected $values = array();
+
+	/**
+	 * GUIDs of plugins for which we will later load settings
+	 *
+	 * @var int[]
+	 */
+	protected $plugin_guids = array();
+
+	/**
+	 * @var Database
+	 */
+	protected $db;
+
+	/**
+	 * Constructor
+	 *
+	 * @param Database $db Elgg Database
+	 */
+	public function __construct(Database $db) {
+		$this->db = $db;
+	}
+
+	/**
+	 * Set the settings for a plugin in the cache
+	 *
+	 * Note this does NOT invalidate any other part of the cache.
+	 *
+	 * @param int   $entity_guid The GUID of the entity
+	 * @param array $values      The settings to cache
+	 * @return void
+	 *
+	 * @access private For testing only
+	 */
+	public function inject($entity_guid, array $values) {
+		$this->values[$entity_guid] = $values;
+	}
+
+	/**
+	 * Get all the non-user, non-internal settings for a plugin.
+	 *
+	 * @param int $entity_guid The GUID of the entity
+	 *
+	 * @return null|string[] null if settings are not cached for this entity
```

Comment:
null = settings not loaded for entity. [] = there are 0 settings. I wanted to eliminate a separate has() method.